backends/ebpf/psa/ebpfPsaControl.cpp

```diff
@@ -32,9 +32,38 @@ void ControlBodyTranslatorPSA::processMethod(const P4::ExternMethod* method) {
         auto counterMap = control->getCounter(name);
         counterMap->to<EBPFCounterPSA>()->emitMethodInvocation(builder, method, this);
         return;
+    } else if (declType->name.name == "Hash") {
+        auto hash = control->to<EBPFControlPSA>()->getHash(name);
+        hash->processMethod(builder, method->method->name.name, method->expr, this);
+        return;
     }
 
     ControlBodyTranslator::processMethod(method);
 }
 
+bool ControlBodyTranslatorPSA::preorder(const IR::AssignmentStatement* a) {
+    if (auto methodCallExpr = a->right->to<IR::MethodCallExpression>()) {
```

Comment:
I assume that the method invocation cannot appear elsewhere (e.g, in an expression).
Otherwise you will just ignore it.